src/DotNet/Writer/IWriterError.cs

```diff
@@ -12,5 +12,13 @@ public interface IWriterError {
 		/// </summary>
 		/// <param name="message">Error message</param>
 		void Error(string message);
+
+		/// <summary>
+		/// Called when an error is detected (eg. a null pointer or other invalid value). The error
+		/// can be ignored but the written data won't be valid.
+		/// </summary>
+		/// <param name="message">Error message</param>
+		/// <param name="args">Optional message arguments</param>
+		void Error(string message, params object[] args);
```

Comment:
This is still a breaking change.

You can create a new `IWriteError2` interface and check for it at runtime.